versioned/storage/cassandra/src/main/java/org/projectnessie/versioned/storage/cassandra/CassandraBackend.java

```diff
@@ -329,16 +334,40 @@ boolean executeCas(BoundStatement stmt) {
   }
 
   ResultSet execute(BoundStatement stmt) {
-    return session.execute(stmt);
+    try {
+      return session.execute(stmt);
+    } catch (DriverException e) {
+      throw unhandledException(e);
+    }
   }
 
   CompletionStage<AsyncResultSet> executeAsync(BoundStatement stmt) {
     return session.executeAsync(stmt);
   }
 
-  void handleDriverException(DriverException e) {
-    if (e instanceof CASWriteUnknownException) {
-      logCASWriteUnknown((CASWriteUnknownException) e);
+  static RuntimeException unhandledException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
```

Comment:
I think you can narrow down to `QueryConsistencyException`.

Reply:
But that would ignore OverloadedException + UnavailableException

Reply:
Yes but for these two exceptions the result is not "unknown": it's actually 100% sure the write did not happen.